pkg/cmd/server/origin/master.go

```diff
@@ -558,7 +559,13 @@ func (c *MasterConfig) InstallUnprotectedAPI(container *restful.Container) ([]st
 
 // initAPIVersionRoute initializes the osapi endpoint to behave similar to the upstream api endpoint
 func initAPIVersionRoute(root *restful.WebService, prefix string, versions ...string) {
-	versionHandler := apiserver.APIVersionHandler(kapi.Codecs, versions...)
+	versionHandler := apiserver.APIVersionHandler(kapi.Codecs, func(req *restful.Request) *unversioned.APIVersions {
+		apiVersionsForDiscovery := unversioned.APIVersions{
+			// TODO: ServerAddressByClientCIDRs: s.getServerAddressByClientCIDRs(req.Request),
```

Comment:
Already on the follow up list

On Wed, Mar 16, 2016 at 8:04 AM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/origin/master.go
> <https://github.com/openshift/origin/pull/7975#discussion_r56321878>:
>
> > @@ -558,7 +559,13 @@ func (c *MasterConfig) InstallUnprotectedAPI(container *restful.Container) ([]st
> >
> >  // initAPIVersionRoute initializes the osapi endpoint to behave similar to the upstream api endpoint
> >  func initAPIVersionRoute(root *restful.WebService, prefix string, versions ...string) {
> > -	versionHandler := apiserver.APIVersionHandler(kapi.Codecs, versions...)
> > +	versionHandler := apiserver.APIVersionHandler(kapi.Codecs, func(req *restful.Request) *unversioned.APIVersions {
> > +		apiVersionsForDiscovery := unversioned.APIVersions{
> > +			// TODO: ServerAddressByClientCIDRs: s.getServerAddressByClientCIDRs(req.Request),
>
> Seems like something we'll want. Followup issue?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7975/files/9ad61a43602f99e9be4d2504b48a13352980ea49..6401e0e510ff9c48b250a7d23444f768dc7be254#r56321878>
>